src/components/settings/Users/EditUser.vue

```diff
@@ -0,0 +1,219 @@
+            </div>
+
+            <button class="btn btn-primary" type="submit" :disabled="savingPassword">
+                <span v-show="savingPassword" class="spinner-border spinner-border-sm" role="status" aria-hidden="true" />
+                {{ $t("Update Password") }}
+            </button>
+        </form>
+
+        <h5 class="mt-5 mb-4 settings-subheading">{{ $t("Permissions") }}</h5>
+        <div class="form-check form-switch">
+            <label class="form-check-label">
+                <input
+                    :checked="active"
+                    class="form-check-input"
+                    style="scale: 1.4; cursor:pointer"
+                    type="checkbox"
+                    :disabled="saving"
+                    @click="debounce(() => { active = !active; save({ active }); })"
+                >
+                <div class="ps-2">{{ $t("Active") }}</div>
+            </label>
+        </div>
+    </template>
+</template>
+
+<script>
+import { useToast } from "vue-toastification";
+const toast = useToast();
+
+/**
+ * Get debounced function
+ * @returns {function} debounced function
+ */
+function Debounce() {
+    let timeout = null;
+
+    /**
+     * exec callback function after delay if no new call to function happens
+     * @param {function} callback function to execute after delay
+     * @param {number} [delay=100] delay before execute the callback if no new call to function happens
+     * @returns {void}
+     */
+    return function (callback, delay = 100) {
+        clearTimeout(timeout);
+        timeout = setTimeout(() => callback(), delay);
+    };
+}
+
+export default {
+    props: {
+        id: {
+            type: String,
+            required: true
+        }
+    },
+
+    data: () => ({
+        loading: false,
+        username: "",
+        saving: false,
+        currentPassword: "",
+        newPassword: "",
+        repeatNewPassword: "",
+        savingPassword: false,
+        active: false
+    }),
+
+    mounted() {
+        this.getUser();
+    },
+
+    methods: {
+        // Used to prevent dbl "click" evt fired by checkbox label
+        debounce: new Debounce(),
+
+        /**
+         * Get user from server
+         * @returns {void}
+         */
+        getUser() {
+            this.loading = true;
+            this.$root.getSocket().emit("getUser", this.id, (res) => {
+                this.loading = false;
+                if (res.ok) {
+                    const { username, active } = res.user;
+
+                    this.username = username;
+                    this.active = active;
+                } else {
+                    toast.error(res.msg);
+                }
+            });
+        },
+
+        /**
+         * Check new passwords match before saving it
+         * @returns {void}
+         */
+        savePassword() {
+            this.savingPassword = true;
+            const { currentPassword, newPassword, repeatNewPassword } = this;
```

Comment:
> why are you extracting this here?

avoid `this.` everywhere.

> (I mostly use the composition API, is this save in the options api?)

This is perfectly safe both in option API and composition API when used inside a function like this because we don't need any reactivity here. We just want the values when the function is called, that's it 😉

Reply:
But they loose reactivity, right?
In this case, I would argue that it would probably be advisable to not do so, as this could lead to future debugging effort.

(feel free to overrule, not a strong opinion)

Reply:
> But they loose reactivity, right?

Not at all. There is no reactivity concerns here. This is just a regular vanilla JS function, part of an object (here a vuejs component instance), which truly don't care about vuejs ref, proxy, whatever...